client/api/python/heketi/heketi.py

```diff
@@ -44,13 +52,156 @@ def _set_token_in_header(self, headers, method, uri):
         token = jwt.encode(claims, self.key, algorithm='HS256')
         headers['Authorization'] = 'bearer ' + token
 
+        return headers
+
 
     def hello(self):
         method = 'GET'
         uri = '/hello'
 
-        headers={}
-        self._set_token_in_header(headers, method, uri)
+        headers = self._set_token_in_header(method, uri)
         r = requests.get(self.host + uri, headers=headers)
         return r.status_code == requests.codes.ok
 
+
+    def _make_request(self, method, uri, data={}, headers={}):
+        headers = self._set_token_in_header(method, uri)
```

Comment:
This overwrites headers if it was passed in.  I think it should append to the dictionary instead:

```python
headers.update(self._set.....())
```